### ISSUE TITLE
If a module has the same name as the containing folder, the generated diagram doesn't work

### ISSUE THREAD

**Issue by gdesantos** [iurysza/module-graph#6]
## 🐛 Describe the bug

If you have a folder with the same name as a submodule, it will not work.
For example:

feature ->
 feature
 core
 domain
 ui

**Comment by iurysza** [comment]
I attempted to group the modules into subgraphs using folders, but it seems I overlooked a potential conflict in cases like the one you mentioned. Maybe it wasn't such a great idea 😆 

I'll dive deeper into this issue and explore the available options to find a solution. Thanks for the report!

**Comment by iurysza** [comment]
Can you provide a branch or repo where I can reproduce this?
I've tried doing it on the `sample` project with this structure:
```txt
+--- Project ':sample:alpha'
+--- Project ':sample:beta'
+--- Project ':sample:container'
|    +--- Project ':sample:container:delta'
|    \--- Project ':sample:container:gama'
+--- Project ':sample:sample'
|    \--- Project ':sample:sample:teta'
\--- Project ':sample:zeta'
```

and it gave me this:
 ![image](https://user-images.githubusercontent.com/4378161/229363493-9383232f-8a93-4c04-9cdb-28b7c94b240e.png)

I think we would need to get the full path of each module and give up the subgraph approach for it to work correctly.
Maybe we can have the option to do that, but I'm not sure if this is such a common use case to justify it yet.
They would look more like [this](https://miro.medium.com/v2/resize:fit:1400/format:webp/1*fg571ukn-dDaEV7FHAyd4A.png) in this case.

**Comment by gdesantos** [comment]
<img width="244" alt="Screenshot 2023-04-04 at 13 29 16" src="https://user-images.githubusercontent.com/3294155/229778232-7059e36f-2101-42b3-b690-0fe527c7e8f0.png">
<img width="593" alt="Screenshot 2023-04-04 at 13 29 24" src="https://user-images.githubusercontent.com/3294155/229778240-a57632a1-98c6-477a-93b7-8a4a860248ea.png">
<img width="1039" alt="Screenshot 2023-04-04 at 13 29 34" src="https://user-images.githubusercontent.com/3294155/229778243-8ea0bc58-d598-49f8-a799-fdc7317fb758.png">

**Comment by iurysza** [comment]
Understood! Mermaid doesn't allow subgraphs to share the same name as nodes, so one workaround is to exclude subgraphs entirely. In your scenario, the resulting graph would look like this:

```mermaid
graph LR  
  app --> myModule:myModule
```

I think we'll need to consider supporting this functionality for situations involving complex folder structures or conflicts of this kind.

I'll come back to this issue. Thanks!

**Comment by ferinagy** [comment]
I think I just ran into a similar issue that has same underlying cause. In our project, we have sometimes modules with a same name, under a different folder, eg. `:cards:api` and `:main:api`:

```
+- :app
+- :cards
|  +- :cards:api
|  \- :cards:fake
+- :main
   +- :main:api
   \- :main:fake
```

With the current logic, only one `api` node is created in the rendered graph:

```
graph LR

  subgraph cards
    api
    fake
  end
  subgraph main
    api
    fake
  end
  app --> cards
  cards --> api %% :cards:api
  app --> main
  main --> api  %% :main:api
```

```mermaid
graph LR

  subgraph cards
    api
    fake
  end
  subgraph main
    api
    fake
  end
  app --> cards
  cards --> api
  app --> main
  main --> api
```

While in the file structure, the `:cards:api` is contained in `:cards`, looking at the modules, `:cards` is same as other modules, so rendering it as a subgraph is IMHO confusing. Also with subgraph, the dependency `:cards --> :cards:api` is almost invisible.

> I think we would need to get the full path of each module and give up the subgraph approach for it to work correctly.

**Comment by iurysza** [comment]
This is the next thing I'll work on. I'm thinking of introducing this behavior as an option, and then turning it into the default behavior when we reach 1.0.

**Comment by iurysza** [comment]
This new feature is available in [v0.5.0](https://github.com/iurysza/module-graph/releases/tag/v0.5.0)
